// @mui/x-date-pickers-pro/legacy/internal/hooks/useDesktopRangePicker/useDesktopRangePicker.js
import _extends from "@babel/runtime/helpers/esm/extends";
import * as React from 'react';
import { resolveComponentProps, useSlotProps } from '@mui/base/utils';
import { useLicenseVerifier } from '@mui/x-license-pro';
import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
import { PickersLayout } from '@mui/x-date-pickers/PickersLayout';
import { executeInTheNextEventLoopTick, getActiveElement, usePicker, PickersPopper } from '@mui/x-date-pickers/internals';
import { useRangePickerInputProps } from '../useRangePickerInputProps';
import { getReleaseInfo } from '../../utils/releaseInfo';
import { useRangePosition } from '../useRangePosition';
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
var releaseInfo = getReleaseInfo();
export var useDesktopRangePicker = function useDesktopRangePicker(_ref) {
  var _slots$layout;
  var props = _ref.props,
    valueManager = _ref.valueManager,
    validator = _ref.validator;
  useLicenseVerifier('x-date-pickers-pro', releaseInfo);
  var slots = props.slots,
    slotProps = props.slotProps,
    className = props.className,
    sx = props.sx,
    format = props.format,
    readOnly = props.readOnly,
    disabled = props.disabled,
    autoFocus = props.autoFocus,
    disableOpenPicker = props.disableOpenPicker,
    localeText = props.localeText;
  var fieldRef = React.useRef(null);
  var popperRef = React.useRef(null);
  var _useRangePosition = useRangePosition(props),
    rangePosition = _useRangePosition.rangePosition,
    onRangePositionChange = _useRangePosition.onRangePositionChange;
  var _usePicker = usePicker({
      props: props,
      valueManager: valueManager,
      wrapperVariant: 'desktop',
      validator: validator,
      autoFocusView: true,
      additionalViewProps: {
        rangePosition: rangePosition,
        onRangePositionChange: onRangePositionChange
      }
    }),
    open = _usePicker.open,
    actions = _usePicker.actions,
    layoutProps = _usePicker.layoutProps,
    renderCurrentView = _usePicker.renderCurrentView,
    shouldRestoreFocus = _usePicker.shouldRestoreFocus,
    pickerFieldProps = _usePicker.fieldProps;
  var handleBlur = function handleBlur() {
    executeInTheNextEventLoopTick(function () {
      var _fieldRef$current, _popperRef$current;
      if ((_fieldRef$current = fieldRef.current) != null && _fieldRef$current.contains(getActiveElement(document)) || (_popperRef$current = popperRef.current) != null && _popperRef$current.contains(getActiveElement(document))) {
        return;
      }
      actions.onDismiss();
    });
  };
  var fieldSlotProps = useRangePickerInputProps({
    wrapperVariant: 'desktop',
    open: open,
    actions: actions,
    readOnly: readOnly,
    disabled: disabled,
    disableOpenPicker: disableOpenPicker,
    localeText: localeText,
    onBlur: handleBlur,
    rangePosition: rangePosition,
    onRangePositionChange: onRangePositionChange
  });
  var Field = slots.field;
  var fieldProps = useSlotProps({
    elementType: Field,
    externalSlotProps: slotProps == null ? void 0 : slotProps.field,
    additionalProps: _extends({}, pickerFieldProps, {
      readOnly: readOnly,
      disabled: disabled,
      className: className,
      sx: sx,
      format: format,
      autoFocus: autoFocus && !props.open,
      ref: fieldRef
    }),
    ownerState: props
  });
  var slotsForField = _extends({
    textField: slots.textField,
    root: slots.fieldRoot,
    separator: slots.fieldSeparator
  }, fieldProps.slots);
  var slotPropsForField = _extends({}, fieldProps.slotProps, {
    textField: function textField(ownerState) {
      var externalInputProps = resolveComponentProps(slotProps == null ? void 0 : slotProps.textField, ownerState);
      return _extends({}, ownerState.position === 'start' ? fieldSlotProps.startInput : fieldSlotProps.endInput, externalInputProps);
    },
    root: function root(ownerState) {
      var externalRootProps = resolveComponentProps(slotProps == null ? void 0 : slotProps.fieldRoot, ownerState);
      return _extends({}, externalRootProps, fieldSlotProps.root);
    },
    separator: slotProps == null ? void 0 : slotProps.fieldSeparator
  });
  var slotPropsForLayout = _extends({}, slotProps, {
    toolbar: _extends({}, slotProps == null ? void 0 : slotProps.toolbar, {
      rangePosition: rangePosition,
      onRangePositionChange: onRangePositionChange
    })
  });
  var Layout = (_slots$layout = slots == null ? void 0 : slots.layout) != null ? _slots$layout : PickersLayout;
  var renderPicker = function renderPicker() {
    return /*#__PURE__*/_jsxs(LocalizationProvider, {
      localeText: localeText,
      children: [/*#__PURE__*/_jsx(Field, _extends({}, fieldProps, {
        slots: slotsForField,
        slotProps: slotPropsForField
      })), /*#__PURE__*/_jsx(PickersPopper, _extends({
        role: "tooltip",
        containerRef: popperRef,
        anchorEl: fieldRef.current,
        onBlur: handleBlur
      }, actions, {
        open: open,
        slots: slots,
        slotProps: slotProps,
        shouldRestoreFocus: shouldRestoreFocus,
        children: /*#__PURE__*/_jsx(Layout, _extends({}, layoutProps, slotProps == null ? void 0 : slotProps.layout, {
          slots: slots,
          slotProps: slotPropsForLayout,
          children: renderCurrentView()
        }))
      }))]
    });
  };
  return {
    renderPicker: renderPicker
  };
};
