// @mui/x-date-pickers-pro/legacy/internal/utils/valueManagers.js
import _toConsumableArray from "@babel/runtime/helpers/esm/toConsumableArray";
import _extends from "@babel/runtime/helpers/esm/extends";
import _slicedToArray from "@babel/runtime/helpers/esm/slicedToArray";
import { replaceInvalidDateByNull, splitFormatIntoSections, addPositionPropertiesToSections, createDateStrForInputFromSections, getSectionOrder as _getSectionOrder } from '@mui/x-date-pickers/internals';
import { splitDateRangeSections, removeLastSeparator } from './date-fields-utils';
export var rangeValueManager = {
  emptyValue: [null, null],
  getTodayValue: function getTodayValue(utils) {
    return [utils.date(), utils.date()];
  },
  cleanValue: function cleanValue(utils, value) {
    return value.map(function (date) {
      return replaceInvalidDateByNull(utils, date);
    });
  },
  areValuesEqual: function areValuesEqual(utils, a, b) {
    return utils.isEqual(a[0], b[0]) && utils.isEqual(a[1], b[1]);
  },
  isSameError: function isSameError(a, b) {
    return b !== null && a[1] === b[1] && a[0] === b[0];
  },
  defaultErrorState: [null, null]
};
export var rangeFieldValueManager = {
  updateReferenceValue: function updateReferenceValue(utils, value, prevReferenceValue) {
    var shouldKeepStartDate = value[0] != null && utils.isValid(value[0]);
    var shouldKeepEndDate = value[1] != null && utils.isValid(value[1]);
    if (!shouldKeepStartDate && !shouldKeepEndDate) {
      return prevReferenceValue;
    }
    if (shouldKeepStartDate && shouldKeepEndDate) {
      return value;
    }
    if (shouldKeepStartDate) {
      return [value[0], prevReferenceValue[0]];
    }
    return [prevReferenceValue[1], value[1]];
  },
  getSectionsFromValue: function getSectionsFromValue(utils, localeText, prevSections, _ref, format) {
    var _ref2 = _slicedToArray(_ref, 2),
      start = _ref2[0],
      end = _ref2[1];
    var prevDateRangeSections = prevSections == null ? {
      startDate: null,
      endDate: null
    } : splitDateRangeSections(prevSections);
    var getSections = function getSections(newDate, prevDateSections, position) {
      var shouldReUsePrevDateSections = !utils.isValid(newDate) && !!prevDateSections;
      if (shouldReUsePrevDateSections) {
        return prevDateSections;
      }
      var sections = splitFormatIntoSections(utils, localeText, format, newDate);
      return sections.map(function (section, sectionIndex) {
        if (sectionIndex === sections.length - 1 && position === 'start') {
          return _extends({}, section, {
            dateName: position,
            endSeparator: "".concat(section.endSeparator, "\u2069 \u2013 \u2066")
          });
        }
        return _extends({}, section, {
          dateName: position
        });
      });
    };
    return addPositionPropertiesToSections([].concat(_toConsumableArray(getSections(start, prevDateRangeSections.startDate, 'start')), _toConsumableArray(getSections(end, prevDateRangeSections.endDate, 'end'))));
  },
  getValueStrFromSections: function getValueStrFromSections(sections) {
    var dateRangeSections = splitDateRangeSections(sections);
    return createDateStrForInputFromSections([].concat(_toConsumableArray(dateRangeSections.startDate), _toConsumableArray(dateRangeSections.endDate)));
  },
  getActiveDateSections: function getActiveDateSections(sections, activeSection) {
    var index = activeSection.dateName === 'start' ? 0 : 1;
    var dateRangeSections = splitDateRangeSections(sections);
    return index === 0 ? removeLastSeparator(dateRangeSections.startDate) : dateRangeSections.endDate;
  },
  parseValueStr: function parseValueStr(valueStr, referenceValue, parseDate) {
    // TODO: Improve because it would not work if the date format has `–` as a separator.
    var _valueStr$split = valueStr.split('–'),
      _valueStr$split2 = _slicedToArray(_valueStr$split, 2),
      startStr = _valueStr$split2[0],
      endStr = _valueStr$split2[1];
    return [startStr, endStr].map(function (dateStr, index) {
      if (dateStr == null) {
        return null;
      }
      return parseDate(dateStr.trim(), referenceValue[index]);
    });
  },
  getActiveDateManager: function getActiveDateManager(utils, state, activeSection) {
    var index = activeSection.dateName === 'start' ? 0 : 1;
    var updateDateInRange = function updateDateInRange(newDate, prevDateRange) {
      return index === 0 ? [newDate, prevDateRange[1]] : [prevDateRange[0], newDate];
    };
    return {
      activeDate: state.value[index],
      referenceActiveDate: state.referenceValue[index],
      getNewValueFromNewActiveDate: function getNewValueFromNewActiveDate(newActiveDate) {
        return {
          value: updateDateInRange(newActiveDate, state.value),
          referenceValue: newActiveDate == null || !utils.isValid(newActiveDate) ? state.referenceValue : updateDateInRange(newActiveDate, state.referenceValue)
        };
      }
    };
  },
  hasError: function hasError(error) {
    return error[0] != null || error[1] != null;
  },
  getSectionOrder: function getSectionOrder(utils, localeText, format, isRTL) {
    var splitedFormat = splitFormatIntoSections(utils, localeText, format, null);
    return _getSectionOrder([].concat(_toConsumableArray(splitedFormat.slice(0, splitedFormat.length - 1)), [_extends({}, splitedFormat[splitedFormat.length - 1], {
      endSeparator: ' – '
    })], _toConsumableArray(splitedFormat)), isRTL);
  }
};
